### PR TITLE
Hide benchmarks behind a feature flag

### LINKED ISSUE #4873: Hide benchmarking pipeline behind a feature flag
By default, a Substrate node should not expose the benchmarking pipeline added by https://github.com/paritytech/substrate/pull/4818

Instead, a user should have to compile their node using a specific flag like `--features=runtime_benchmarks`.

Without this feature flag, we should hide:

* The bechmark host functions (to prevent misuse within the runtime environment)
* The benchmark runtime api (they wont work)
* The benchmark runtime code (to keep wasm code size small)

### PR DESCRIPTION
**Pull request by DarkEld3r**
Closes #4873.

### CODE REVIEW

File: primitives/api/proc-macro/src/impl_runtime_apis.rs [hunk 1]
```diff
@@ -163,7 +163,7 @@ fn generate_impl_calls(
 				)?;
 
 				impl_calls.push(
-					(impl_trait_ident.clone(), method.sig.ident.clone(), impl_call)
+					(impl_trait_ident.clone(), method.sig.ident.clone(), impl_call, impl_.attrs.clone())
```
Comment (bkchr) [inline primitives/api/proc-macro/src/impl_runtime_apis.rs]: I think the best would be to filter the attributes and only propagate `cfg(something)` for now.

File: primitives/api/proc-macro/src/impl_runtime_apis.rs [hunk 2]
```diff
@@ -745,3 +759,32 @@ fn impl_runtime_apis_impl_inner(api_impls: &[ItemImpl]) -> Result<TokenStream> {
 		)
 	)
 }
+
+// Filters all attributes except the cfg ones.
+fn filter_attrs(attrs: &[Attribute]) -> Vec<Attribute> {
```
Comment (marcio-diaz) [inline primitives/api/proc-macro/src/impl_runtime_apis.rs]:
```suggestion
fn filter_cfg_attrs(attrs: &[Attribute]) -> Vec<Attribute> {
```

File: bin/node/runtime/Cargo.toml [hunk 3]
```diff
@@ -132,3 +132,4 @@ std = [
 	"pallet-recovery/std",
 	"pallet-vesting/std",
 ]
+runtime_benchmarks = ["frame-benchmarking"]
```
Comment (expenses) [inline bin/node/runtime/Cargo.toml]: For consistency, I'd prefer this feature to be called `runtime-benchmarks`.

Comment (DarkEld3r) [reply bin/node/runtime/Cargo.toml]: Makes sense.

**Comment by bkchr** [comment]
Sorry for the delay: https://github.com/paritytech/substrate/pull/5131

**Pull request merged**